lib/remap-lang/src/function/split.rs

```diff
@@ -0,0 +1,90 @@
+use crate::{
+    expression::Error, Argument, ArgumentList, Expr, Expression, Function, Object, Parameter,
+    Result, State, Value,
+};
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug)]
+pub(crate) struct Split;
+
+impl Function for Split {
+    fn identifier(&self) -> &'static str {
+        "split"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::String(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "pattern",
+                accepts: |v| matches!(v, Value::String(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "case_sensitive",
+                accepts: |v| matches!(v, Value::Boolean(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required_expr("value")?;
+        let pattern = arguments.required("pattern")?;
+        let limit = arguments.optional_expr("limit")?;
+
+        Ok(Box::new(SplitFn {
+            value,
+            pattern,
+            limit,
+        }))
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct SplitFn {
+    value: Box<Expr>,
+    pattern: Argument,
+    limit: Option<Box<Expr>>,
+}
+
+impl Expression for SplitFn {
+    fn execute(&self, state: &mut State, object: &mut dyn Object) -> Result<Option<Value>> {
+        let string: String = self
+            .value
+            .execute(state, object)?
+            .ok_or(Error::Missing)?
+            .try_into()?;
+
```

Comment:
Is there any reason why these can't be `required` and `optional` macros as before?

Reply:
No reason, but as mentioned in the comments, this function will be moved back to vector in a follow-up PR, where the `required` and `optional` macro's currently live.